pkg/build/build.go

```diff
@@ -554,30 +554,110 @@ func WithEnvFile(envFile string) Option {
 	}
 }
 
-// Load the configuration data from the build context configuration file.
-func (cfg *Configuration) Load(ctx Context) error {
-	data, err := os.ReadFile(ctx.ConfigFile)
-	if err != nil {
-		return fmt.Errorf("unable to load configuration file: %w", err)
+type ConfigurationParsingOption func(*configOptions)
+
+type configOptions struct {
+	abstractFS     fs.FS
+	configFilePath string
+	envFilePath    string
+	logger         Logger
+}
+
+// include reconciles all given opts into the receiver variable, such that it is
+// ready to use for config parsing.
+func (options *configOptions) include(opts ...ConfigurationParsingOption) {
+	for _, fn := range opts {
+		fn(options)
+	}
+	if options.logger == nil {
+		options.logger = nopLogger{}
+	}
+}
+
+// open handles opening of a file with the given name, in such a way that
+// automatically chooses to use an abstract or concrete OS filesystem.
+func (options *configOptions) open(filepath string) (fs.File, error) {
+	if filesystem := options.abstractFS; filesystem != nil {
```

Comment:
Instead of switching here can we make the default fs a real filesystem fs?

Reply:
👍